includes/actions/update-next-payment.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace AutomateWoo_Subscriptions;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Change a subscription's next payment date.
+ *
+ *
+ * @class Action_Subscription_Update_Next_Payment
+ * @since 1.0.0
+ */
+class Action_Subscription_Update_Next_Payment extends Abstract_Action_Subscription {
+
+
+	/**
+	 * Explain to store admin what this action does via a unique title and description.
+	 */
+	public function load_admin_details() {
+		parent::load_admin_details();
+		$this->title       = __( 'Update Next Payment', 'automatewoo-subscriptions' );
+		$this->description = __( 'Change a subscription\'s next payment date.', 'automatewoo-subscriptions' );
+	}
+
+	/**
+	 * Edit the item managed by this class on the subscription passed in the workflow's trigger
+	 *
+	 * @throws \Exception When there is an error.
+	 */
+	public function run() {
+		$subscription     = $this->get_subscription_to_edit();
+		$old_payment_date = $subscription->get_date( 'next_payment' );
+		$subscription->update_meta_data( '_old_schedule_next_payment', $old_payment_date );
+		$subscription->save();
```

Comment:
Let's not save the original schedule next payment in the meta. Create a private class variable `$original_payment_date` which can be used when adding the subscription note.
```suggestion
		$subscription                            = $this->get_subscription_to_edit();
		$this->original_payment_date = $subscription->get_date( 'next_payment' );
```